internal/resources/azure/storage_table.go

```diff
@@ -132,66 +125,44 @@ func (r *StorageTable) dataStorageCostComponent() *schema.CostComponent {
 	}
 }
 
-func (r *StorageTable) operationsCostComponents() []*schema.CostComponent {
-	costComponents := []*schema.CostComponent{}
-
-	if !contains([]string{"GZRS", "RA-GZRS"}, strings.ToUpper(r.AccountReplicationType)) {
-		var class1Qty *decimal.Decimal
-		if r.MonthlyClass1Operations != nil {
-			class1Qty = decimalPtr(decimal.NewFromInt(*r.MonthlyClass1Operations).Div(decimal.NewFromInt(10000)))
-		}
-
-		costComponents = append(costComponents, &schema.CostComponent{
-			Name:            "Class 1 operations",
-			Unit:            "10k operations",
-			UnitMultiplier:  decimal.NewFromInt(1),
-			MonthlyQuantity: class1Qty,
-			ProductFilter: &schema.ProductFilter{
-				VendorName:    strPtr("azure"),
-				Region:        strPtr(r.Region),
-				Service:       strPtr("Storage"),
-				ProductFamily: strPtr("Storage"),
-				AttributeFilters: []*schema.AttributeFilter{
-					{Key: "productName", Value: strPtr(r.productName())},
-					{Key: "skuName", Value: strPtr(fmt.Sprintf("Standard %s", strings.ToUpper(r.AccountReplicationType)))},
-					{Key: "meterName", ValueRegex: regexPtr("Class 1 Operations$")},
-				},
-			},
-			PriceFilter: &schema.PriceFilter{
-				PurchaseOption:   strPtr("Consumption"),
-				StartUsageAmount: strPtr("0"),
-			},
-			UsageBased: true,
-		})
+func (r *StorageTable) operationsCostComponent(operation string, usage *int64) *schema.CostComponent {
+	var qty *decimal.Decimal
+	if usage != nil {
+		qty = decimalPtr(decimal.NewFromInt(*usage).Div(decimal.NewFromInt(10000)))
 	}
 
-	var class2Qty *decimal.Decimal
-	if r.MonthlyClass2Operations != nil {
-		class2Qty = decimalPtr(decimal.NewFromInt(*r.MonthlyClass2Operations).Div(decimal.NewFromInt(10000)))
+	// RA-GZRS is a special case. There's no listed prices for the write operations, but they
+	// use the same price as the read operations
+	operationLookup := operation
+	if !r.HasCustomerManagedKey && strings.EqualFold(r.AccountReplicationType, "ra-gzrs") && (operation == "Batch Write" || operation == "Write") {
+		operationLookup = "Read"
 	}
 
-	costComponents = append(costComponents, &schema.CostComponent{
-		Name:            "Class 2 operations",
+	if operationLookup == "Write" {
+		// Ensure we don't look up prices for Batch Write
+		operationLookup = "(?<!Batch\\s+)Write"
```

Comment:
When we look up '*Write Operations' in the pricing API we want to make sure we don't match on 'Batch Write Operations' since those have different prices, so this makes sure the 'Batch ' part isn't matched.